crates/extensions/c8y_mapper_ext/src/serializer.rs

```diff
@@ -116,16 +129,53 @@ impl MeasurementVisitor for C8yJsonSerializer {
         Ok(())
     }
 
-    fn visit_measurement(&mut self, key: &str, value: f64) -> Result<(), Self::Error> {
-        self.json.write_key(key)?;
-
+    fn visit_other_fragments(&mut self, name: &str, value: &str) -> Result<(), Self::Error> {
         if self.is_within_group {
-            self.write_value_obj(value)?;
-        } else {
-            self.json.write_open_obj();
-            self.json.write_key(key)?;
-            self.write_value_obj(value)?;
-            self.json.write_close_obj();
+            return Err(MeasurementStreamError::UnexpectedType.into());
+        }
+        match name {
+            "type" => {
+                self.json.write_key("type")?;
+                self.json.write_str(value)?;
+
+                self.type_present = true;
+            }
+
+            "externalSource" => {
+                return Err(C8yJsonSerializationError::InvalidMeasurementName {
+                    name: name.to_string(),
+                });
```

Comment:
I would consider to introduce a new error case as the name is not invalid in this context but unexpected.

```suggestion
                return Err(C8yJsonSerializationError::UnexpectedExternalSource);
```